// react-util/build/check.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.uncheckedVals = uncheckedVals;
exports.checkedVals = checkedVals;
exports.uncheckRefs = uncheckRefs;
exports.checkRefs = checkRefs;
exports.uncheck = uncheck;
exports.check = check;

/**
 * check
 */

function uncheckedVals(pattern) {
  var _this = this;

  var vals = [];

  Object.keys(this.refs).forEach(function (ref) {
    var dom = _this.refs[ref];
    if (match(pattern, ref) && dom.checked === false) {
      vals.push(dom.value);
    }
  });

  return vals;
}

function checkedVals(pattern) {
  var _this2 = this;

  var vals = [];

  Object.keys(this.refs).forEach(function (ref) {
    var dom = _this2.refs[ref];
    if (match(pattern, ref) && dom.checked) {
      vals.push(dom.value);
    }
  });

  return vals;
}

function uncheckRefs(pattern) {
  var _this3 = this;

  Object.keys(this.refs).forEach(function (ref) {
    if (match(pattern, ref)) {
      _this3.refs[ref].checked = false;
    }
  });
}

function checkRefs(pattern) {
  var _this4 = this;

  Object.keys(this.refs).forEach(function (ref) {
    if (match(pattern, ref)) {
      _this4.refs[ref].checked = true;
    }
  });
}

function uncheck(ref) {
  var dom = this.refs[ref];
  if (dom) {
    dom.checked = false;
  }
}

function check() {
  var dom = this.refs[ref];
  if (dom) {
    dom.checked = true;
  }
}
