// react-util/build/form.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Radio = exports.Checkbox = exports.Textarea = exports.Input = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function noop() {}

var Input = exports.Input = function (_Component) {
  _inherits(Input, _Component);

  function Input() {
    _classCallCheck(this, Input);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(Input).apply(this, arguments));
  }

  _createClass(Input, [{
    key: 'render',
    value: function render() {
      var _props = this.props;
      var _props$onChange = _props.onChange;

      var _onChange = _props$onChange === undefined ? noop : _props$onChange;

      var value = _props.value;


      return React.createElement('input', _extends({}, this.props, { value: value,
        onChange: function onChange(e) {
          return _onChange(e.target.value, e);
        } }));
    }
  }]);

  return Input;
}(_react.Component);

Input.propTypes = {
  onChange: _react.PropTypes.func,
  value: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number])
};

var Textarea = exports.Textarea = function (_Component2) {
  _inherits(Textarea, _Component2);

  function Textarea() {
    _classCallCheck(this, Textarea);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(Textarea).apply(this, arguments));
  }

  _createClass(Textarea, [{
    key: 'render',
    value: function render() {
      var _props2 = this.props;
      var _props2$onChange = _props2.onChange;

      var _onChange2 = _props2$onChange === undefined ? noop : _props2$onChange;

      var value = _props2.value;


      return React.createElement('textarea', _extends({}, this.props, { value: value,
        onChange: function onChange(e) {
          return _onChange2(e.target.value, e);
        } }));
    }
  }]);

  return Textarea;
}(_react.Component);

Textarea.propTypes = {
  onChange: _react.PropTypes.func,
  value: _react.PropTypes.string
};

var Checkbox = exports.Checkbox = function (_Component3) {
  _inherits(Checkbox, _Component3);

  function Checkbox() {
    _classCallCheck(this, Checkbox);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(Checkbox).apply(this, arguments));
  }

  _createClass(Checkbox, [{
    key: 'render',
    value: function render() {
      var _props3 = this.props;
      var _props3$onChange = _props3.onChange;

      var _onChange3 = _props3$onChange === undefined ? noop : _props3$onChange;

      var checked = _props3.checked;


      return React.createElement('input', { type: 'checkbox', checked: checked,
        onChange: function onChange(e) {
          return _onChange3(e.target.checked, e);
        } });
    }
  }]);

  return Checkbox;
}(_react.Component);

Checkbox.propTypes = {
  onChange: _react.PropTypes.func,
  checked: _react.PropTypes.bool
};

var Radio = exports.Radio = function (_Component4) {
  _inherits(Radio, _Component4);

  function Radio() {
    _classCallCheck(this, Radio);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(Radio).apply(this, arguments));
  }

  _createClass(Radio, [{
    key: 'render',
    value: function render() {
      var _props4 = this.props;
      var _props4$onChange = _props4.onChange;

      var _onChange4 = _props4$onChange === undefined ? noop : _props4$onChange;

      var checked = _props4.checked;


      return React.createElement('input', { type: 'radio', checked: checked,
        onChange: function onChange(e) {
          return _onChange4(e.target.checked, e);
        } });
    }
  }]);

  return Radio;
}(_react.Component);

Radio.propTypes = {
  onChange: _react.PropTypes.func,
  checked: _react.PropTypes.bool
};
